### PR TITLE
Check whether schemas exist or not before running pegasusSchemaSnapshotCheck task

### PR DESCRIPTION
**Pull request by nickibi**
Check whether schemas exist or not before running pegasusSchemaSnapshotCheck task

### CODE REVIEW

File: gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java [hunk 1]
```diff
@@ -1441,8 +1447,8 @@ protected void configurePegasusSchemaSnapshotGeneration(Project project, SourceS
           task.dependsOn(generatePegasusSchemaSnapshot);
           task.setCurrentSnapshotDirectory(publishablePegasusSchemaSnapshotDir.toFile());
           task.setPreviousSnapshotDirectory(pegasusSchemaSnapshotDir);
-          task.setCodegenClasspath(project.getConfigurations() .getByName(SCHEMA_ANNOTATION_HANDLER_CONFIGURATION)
-              .plus(project.getConfigurations().getByName(PEGASUS_PLUGIN_CONFIGURATION))
+          task.setCodegenClasspath(project.getConfigurations().getByName(PEGASUS_PLUGIN_CONFIGURATION)
```
Comment (evanw555) [inline gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java:1450]: Does changing the ordering here fix the problem we talked about earlier (related to the wrong pegasus version being picked up in the schema annotation handler configuration)?

File: gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java [hunk 2]
```diff
@@ -850,7 +850,13 @@ public void apply(Project project)
         configurePegasusSchemaSnapshotGeneration(project, sourceSet, false);
       }
 
-      configurePegasusSchemaSnapshotGeneration(project, sourceSet, true);
+      // Do compatibility check if there are extension schemas.
+      if (!SharedFileUtils.getSuffixedFiles(project,  project.file(getExtensionSchemaPath(project, sourceSet)),
```
Comment (evanw555) [inline gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java:854]: Looking at other methods like `configureRestClientConfiguration`, this sort of conditional check is done inside the method. I'd suggest making it consistent.

File: gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java (hunk 1 above)
Comment (nickibi) [reply gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java:1450]: Yes, it fixes the problem

**Review by evanw555 — APPROVED** [review]
LGTM

**Pull request merged**